### PR TITLE
Firma 363 rol identification

### PR DESCRIPTION
**Pull request by SergioPH**

### CODE REVIEW

File: src/Network/UserRolesNetwork.m [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                        (unsigned long)operation,datParameter, [data base64EncodedString]];
+    
+    if ([[LoginService instance] sessionId]){
+        params = [NSString stringWithFormat:@"%@%@", params , [NSString stringWithFormat:@"&ssid=%@", [[LoginService instance] sessionId]]];
+    }
+    
+    NSData *postData = [params dataUsingEncoding:NSUTF8StringEncoding allowLossyConversion:YES];
+    NSString *postLength = [NSString stringWithFormat:@"%lu", (unsigned long)[params length]];
+    
+//    NSString *baseURL = [NSString stringWithFormat:@"%@?%@", SERVER_URL, params];
```
Comment (ggdiez) [inline src/Network/UserRolesNetwork.m:38]: Remove this commented line please

File: src/Network/UserRolesNetwork.m [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
+    [request setURL:[NSURL URLWithString:baseURL]];
+    [request setHTTPMethod:@"POST"];
+    [request setValue:postLength forHTTPHeaderField:@"Content-Length"];
+    [request setHTTPBody:postData];
+    [request setTimeoutInterval:30.0];
```
Comment (ggdiez) [inline src/Network/UserRolesNetwork.m:45]: Timeout better in a constant

File: src/core/business/ValidateController.m [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+@implementation ValidateController
+
+@synthesize dataSource = _dataSource;
+
+#pragma mark - Init methods
+
+- (ValidateController *)initXMLParser {
+    self = [super init];
+    // init array of user objects
```
Comment (ggdiez) [inline src/core/business/ValidateController.m:25]: Remove this comment, the code in understood

File: src/ui/controllers/lists/filter/FilterVC.m [hunk 4]
```diff
@@ -231,13 +241,44 @@ - (void)updateContentOffsetForHeight:(CGFloat)height
...
+}
+
 #pragma mark - User Interaction
 
 - (IBAction)didClickCancelButton:(id)sender
 {
-   dispatch_async(dispatch_get_main_queue(), ^{
-	   [self dismissViewControllerAnimated:YES completion:nil];
-	});
+    if ([[UIDevice currentDevice].model isEqualToString:@"iPhone"]) {
```
Comment (ggdiez) [inline src/ui/controllers/lists/filter/FilterVC.m:275]: iPhone should be a constant

File: src/ui/controllers/settings/SettingsVC.m [hunk 5]
```diff
@@ -178,7 +190,43 @@ - (BOOL)shouldPerformSegueWithIdentifier:(NSString *)identifier sender:(id)sende
 			[[LoginService instance] loginWithCertificate:^{
 				segue = YES;
 				dispatch_async(dispatch_get_main_queue(), ^{
-					[self performSegueWithIdentifier:identifier sender:self];
+                    [[UserRolesService instance] getUserRoles:^(NSDictionary *content) {
+                        printf("Success");
```
Comment (ggdiez) [inline src/ui/controllers/settings/SettingsVC.m:194]: Remove this print

File: src/ui/controllers/settings/SettingsVC.m [hunk 6]
```diff
@@ -178,7 +190,43 @@ - (BOOL)shouldPerformSegueWithIdentifier:(NSString *)identifier sender:(id)sende
...
+                                //User with no roles
+                                //continue as always
+                                segue = YES;
+                                dispatch_async(dispatch_get_main_queue(), ^{
+                                    [self performSegueWithIdentifier:identifier sender:self];
+                                });
+                            } else {
+                                //User with roles
+                            [self setRolesInLocalStorage: responseUserRolesDict];
+//                                Navegar a la pantalla de selección de rol
```
Comment (ggdiez) [inline src/ui/controllers/settings/SettingsVC.m:217]: This comment should be in english like the others

File: src/Network/UserRolesNetwork.m (hunk 1 above)
Comment (SergioPH) [reply src/Network/UserRolesNetwork.m:38]: Done

File: src/core/business/ValidateController.m (hunk 3 above)
Comment (SergioPH) [reply src/core/business/ValidateController.m:25]: Done

File: src/ui/controllers/settings/SettingsVC.m (hunk 5 above)
Comment (SergioPH) [reply src/ui/controllers/settings/SettingsVC.m:194]: Done

File: src/ui/controllers/settings/SettingsVC.m (hunk 6 above)
Comment (SergioPH) [reply src/ui/controllers/settings/SettingsVC.m:217]: Done

File: src/Network/UserRolesNetwork.m (hunk 2 above)
Comment (SergioPH) [reply src/Network/UserRolesNetwork.m:45]: Done in more places

File: src/ui/controllers/lists/filter/FilterVC.m (hunk 4 above)
Comment (SergioPH) [reply src/ui/controllers/lists/filter/FilterVC.m:275]: Done in more places

**Pull request merged**